src/shared/watchdog.c

```diff
@@ -29,6 +30,40 @@ static unsigned saturated_usec_to_sec(usec_t val) {
         return (unsigned)(t >= WATCHDOG_TIMEOUT_MAX_SEC ? WATCHDOG_TIMEOUT_MAX_SEC : t); /* Saturate to watchdog max */
 }
 
+static int update_pretimeout(void) {
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY || watchdog_timeout == USEC_INFINITY)
+                return 0;
+        else if (saturated_usec_to_sec(watchdog_pretimeout) >= saturated_usec_to_sec(watchdog_timeout)) {
```

Comment:
i'd drop the "else" here, it's redundant, given that you already left the function already if the first if check held

Reply:
Sure. Do you think it makes sense to do the same for the next `else` as well?